### PR TITLE
PERF-4881 Add genny workload targeting block processing

### PR DESCRIPTION
**Pull request by parker-felix**
**Jira Ticket:** [PERF-4881](https://jira.mongodb.org/browse/PERF-4881)

**Whats Changed:**  
Created a workload that runs against uniformly random timeseries data to target SBE block processing.

**Patch testing results:**  
[Evergreen patch](https://spruce.mongodb.com/version/654d26803e8e86df96dc9eec/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

**Workload Submission form:**  
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)
Form submitted!

**Related PRs:**

### CODE REVIEW

File: src/workloads/query/TimeseriesBlockProcessing.yml [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+  ineffective. This stresses turning buckets into blocks and running block based operations on them.
+
+Keywords:
+  - timeseries
+  - aggregate
+
+GlobalDefaults:
+  Database: &database test
+  Collection: &collection Collection0
+  DocumentCount: &documentCount 1e6
```
Comment (borosaurus) [inline src/workloads/query/TimeseriesBlockProcessing.yml:16]: Can we make it 10m? A lot of time series workloads have around 100m or more so I just want to be somewhere close

File: src/workloads/query/TimeseriesBlockProcessing.yml [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  Repeat: &repeat 100
+  Threads: &threads 1
+  MaxPhases: &maxPhases 11
+  MetaCount: &metaCount 10
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 4000 # With 8 threads we need a larger pool.
```
Comment (borosaurus) [inline src/workloads/query/TimeseriesBlockProcessing.yml:25]: Can probably delete this if we only have one thread

File: src/workloads/query/TimeseriesBlockProcessing.yml [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+                                  { $lt: { ^Date: "2022-01-02T00:00:00" } },
+                              },
+                              { "obj.measurement2": { $gte: 25 } },
+                            ],
+                        },
+                    },
+                    { $count: "count" },
+                  ]
+
+  # TODO: The $or $eq on obj.measurement1 gets converted to a $in so this query can't run in
```
Comment (borosaurus) [inline src/workloads/query/TimeseriesBlockProcessing.yml:270]: I'd throw a reference to SERVER-83057 here. After that ticket goes in we can uncomment it, just so we're getting results. Even if it's not using BP it's useful to see where SBE is regressing

File: src/workloads/query/TimeseriesBlockProcessing.yml [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+  #                             },
+  #                             { "obj.measurement2": { $lte: 175 } },
+  #                             { "obj.measurement3": { $gt: 100, $lt: 800 } },
+  #                           ],
+  #                       },
+  #                   },
+  #                   { $count: "count" },
+  #                 ]
+
+  # The $match will be handled by block processing, but the pipeline from $group onwards will be
```
Comment (borosaurus) [inline src/workloads/query/TimeseriesBlockProcessing.yml:313]: [nit] remove this comment since it will probably get stale soon

**Review by borosaurus — APPROVED** [review]
LGTM

File: src/workloads/query/TimeseriesBlockProcessing.yml [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+    Phases:
+      OnlyActiveInPhases:
+        Active: [10]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: *repeat
+          Database: *database
+          Collection: *collection
+          Operations:
+            - OperationMetricsName:
```
Comment (rtimmons) [inline src/workloads/query/TimeseriesBlockProcessing.yml:328]: (Drive-by review)

Are you intentionally leaving the `OperationMetricsName` fields empty?
I'm not sure of the intended behavior. If this is intentional, a comment would probably help.

Comment (parker-felix) [reply src/workloads/query/TimeseriesBlockProcessing.yml:328]: There wasn't any particular reason for this, think I just copied this pattern from an existing workload. Is there a particular convention for what `OperationMetricsName` should be?

File: src/workloads/query/TimeseriesBlockProcessing.yml (hunk 1 above)
Comment (parker-felix) [reply src/workloads/query/TimeseriesBlockProcessing.yml]: Done. Reduced number of repeats from 100 to 20 since most queries are ~10x longer.

**Review by rtimmons — APPROVED** [review]

**Comment by parker-felix** [comment]
evergreen merge
